### PR TITLE
Record output column datatype in EventListener

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java [hunk 1]
```diff
@@ -118,7 +118,7 @@ protected QueryRunner createQueryRunner()
                         })
                         .withGetViews((connectorSession, prefix) -> {
                             ConnectorViewDefinition definition = new ConnectorViewDefinition(
-                                    "select nationkey as test_column from tpch.tiny.nation",
+                                    "select nationkey AS test_column from tpch.tiny.nation",
```
Comment (ksobolew) [inline testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:124]: I guess you want to capitalize `SELECT` and `FROM` as well

**Review by ksobolew — APPROVED** [review]
Simple enough; but I'd like to see in a commit description what's the purpose of the extra information and what it would be used for (in Atlas)

File: testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java [hunk 2]
```diff
@@ -149,7 +149,7 @@ protected QueryRunner createQueryRunner()
                         })
                         .withColumnMask((schemaTableName, columnName) -> {
                             if (schemaTableName.getTableName().equals("test_table_with_column_mask") && columnName.equals("test_varchar")) {
-                                return new ViewExpression("user", Optional.of("tpch"), Optional.of("tiny"), "(SELECT cast(max(orderkey) AS VARCHAR) FROM orders)");
+                                return new ViewExpression("user", Optional.of("tpch"), Optional.of("tiny"), "(SELECT cast(max(orderkey) AS VARCHAR(15)) FROM orders)");
```
Comment (kokosing) [inline testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java:155]: nit: `VARCHAR` is not a keyword so it should be lower cased.

File: core/trino-spi/src/main/java/io/trino/spi/eventlistener/OutputColumnMetadata.java [hunk 3]
```diff
@@ -23,12 +23,14 @@
 public class OutputColumnMetadata
 {
     private final String columnName;
+    private final String columnDataType;
```
Comment (kokosing) [inline core/trino-spi/src/main/java/io/trino/spi/eventlistener/OutputColumnMetadata.java:26]: columnType?

**Review by kokosing — APPROVED** [review]

**Comment by Praveen2112** [comment]
@ksobolew , @kokosing AC

**Pull request merged**